drivers/gpio/gpio_nrfx.c

```diff
@@ -328,19 +315,26 @@ static int gpio_nrfx_pin_interrupt_configure(struct device *port,
 		return -ENOTSUP;
 	}
 
-	WRITE_BIT(data->pin_int_en, pin, flags & GPIO_INT_ENABLE);
-	WRITE_BIT(data->int_en, pin, true);
-	WRITE_BIT(data->trig_edge, pin, flags & GPIO_INT_EDGE);
-	WRITE_BIT(data->double_edge, pin, (flags & GPIO_INT_LOW_0) &&
-					  (flags & GPIO_INT_HIGH_1));
+	bool pin_int_en = true;
+	bool edge = true;
+	bool active = true;
 
-	bool active_high = ((flags & GPIO_INT_HIGH_1) != 0);
+	if (mode == GPIO_INT_MODE_DISABLED)
+		pin_int_en = false;
 
-	if (((flags & GPIO_INT_LEVELS_LOGICAL) != 0) &&
-	    ((data->general.invert & BIT(pin)) != 0)) {
-		active_high = !active_high;
-	}
-	WRITE_BIT(data->int_active_level, pin, active_high);
+	if (trig == GPIO_INT_TRIG_BOTH)
+		WRITE_BIT(data->double_edge, pin, true);
+
+	if (mode == GPIO_INT_MODE_EDGE)
+		edge = false;
+
+	if (trig == GPIO_INT_TRIG_LOW)
+		active = false;
+
+	WRITE_BIT(data->pin_int_en, pin, pin_int_en);
+	WRITE_BIT(data->int_en, pin, true);
+	WRITE_BIT(data->trig_edge, pin, edge);
+	WRITE_BIT(data->int_active_level, pin, active);
```

Comment:
Doesn't pass the 1 pin testcase. Following version does:
```
static int gpio_nrfx_pin_interrupt_configure(struct device *port,
		unsigned int pin, enum gpio_int_mode mode,
	       	enum gpio_int_trig trig)
{
	struct gpio_nrfx_data *data = get_port_data(port);
	u32_t abs_pin = NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num, pin);

	if ((mode == GPIO_INT_MODE_EDGE) &&
	    (nrf_gpio_pin_dir_get(abs_pin) == NRF_GPIO_PIN_DIR_OUTPUT)) {
		/*
		 * The pin's output value as specified in the GPIO will be
		 * ignored as long as the pin is controlled by GPIOTE.
		 * Pin with output enabled cannot be used as an edge interrupt
		 * source.
		 */
		return -ENOTSUP;
	}

	WRITE_BIT(data->pin_int_en, pin, mode != GPIO_INT_MODE_DISABLED);
	WRITE_BIT(data->int_en, pin, mode != GPIO_INT_MODE_DISABLED);
	WRITE_BIT(data->trig_edge, pin, mode == GPIO_INT_MODE_EDGE);
	WRITE_BIT(data->double_edge, pin, trig == GPIO_INT_TRIG_BOTH);
	WRITE_BIT(data->int_active_level, pin, trig == GPIO_INT_TRIG_HIGH);

	return gpiote_pin_int_cfg(port, pin);
}
```

Reply:
Done.